### ISSUE TITLE
Please keep NetMQ release synchronized with AsyncIO releases

### ISSUE THREAD

**Issue by mfeemster** [zeromq/netmq#467]
If one gets the latest NetMQ, it references AsyncIO 1.17.

Then, when they update their nuget pacakges, they get AsyncIO 1.18

From that point on, app config must have the following inserted into it:

```
<runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="AsyncIO" publicKeyToken="44a94435bd6f33f8" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-0.1.18.0" newVersion="0.1.18.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
```

This becomes a pain if you are managing many applications.

Can you please make NetMQ always reference whatever the latest version of AsyncIO is now that they are basically a pair? This will alleviate the need for any special configuration and will get NetMQ back to a state of "just working" out of the box as it did before the split.

Thanks.

**Comment by drewnoakes** [comment]
Are you using the latest stable, or the latest prerelease version? Prerelease does reference AsyncIO 0.1.18.

At any rate, why would you update AsyncIO? Are you using it outside of NetMQ? It's a (known) weakness of NuGet that user package lists include transitive dependencies. The [upcoming project structure](https://github.com/aspnet/Home/wiki/Project.json-file) (currently in use for ASPNET Core) removes the need for this, so this problem shouldn't occur in future.

**Comment by mfeemster** [comment]
I am using the latest stable.

I updated AsyncIO just because I figured it would be best to get the latest.

Do you have any idea when the pre-release version of NetMQ will transition to stable?

**Comment by gtemgoua** [comment]
I guess mfeemster is having the same issues I have.

**Comment by somdoron** [comment]
Yes, we will fix this in next stable release

**Comment by valentinepeter** [comment]
Hmmm... I suspect this is why my ability to debug code containing NETMQ is totally hosed right now... while the code runs compiled, ever since I updated NuGet and it loaded 0.1.18 and I had reload 0.1.17 when I try to debug I get: 

Error:System.IO.FileLoadException: Could not load file or assembly 'AsyncIO, Version=0.1.18.0, Culture=neutral, PublicKeyToken=44a94435bd6f33f8' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'AsyncIO, Version=0.1.18.0, Culture=neutral, PublicKeyToken=44a94435bd6f33f8' ---> System.IO.FileLoadException: Could not load file or assembly 'AsyncIO, Version=0.1.17.0, Culture=neutral, PublicKeyToken=44a94435bd6f33f8' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'AsyncIO, Version=0.1.17.0, Culture=neutral, PublicKeyToken=44a94435bd6f33f8'

**Comment by sharpe5** [comment]
Probably caused by a mixture of assembly versions in the project. 

To fix, try adding an assembly binding redirect:

http://stackoverflow.com/questions/24848184/why-system-windows-interactivity-doesnt-want-to-work

**Comment by SemtexCharlie** [comment]
I hit this on an older winforms project that has been upgraded to a newer .NET version but just needed to enable auto generation of binding redirects:

https://msdn.microsoft.com/en-us/library/2fc472t2(v=vs.110).aspx

**TL;DR:** Add this element to the first <PropertyGroup> of your csproj:

`<AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>`

**Comment by SpecialHias** [comment]
This is still not fixed right? I get into the same troubles with NetMQ version 4.0.0.1 and AsyncIO 0.1.40.
